tests/golden_config_infra/test_config_reload_with_rendered_golden_config.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+import logging
+import json
+import os
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.config_reload import config_reload, config_reload_minigraph_with_rendered_golden_config_override
+from tests.override_config_table.utilities import backup_config, restore_config, get_running_config, \
+    compare_dicts_ignore_list_order, NON_USER_CONFIG_TABLES
+from tests.common.utilities import update_pfcwd_default_state
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('t0', 't1', 'any'),
+    pytest.mark.disable_loganalyzer,
+]
+
+GOLDEN_CONFIG = "/etc/sonic/golden_config_db.json"
+GOLDEN_CONFIG_BACKUP = "/etc/sonic/golden_config_db.json_before_override"
+
+
+def file_exists_on_dut(duthost, filename):
+    return duthost.stat(path=filename).get('stat', {}).get('exists', False)
+
+
+@pytest.fixture(scope="module")
+def setup_env(duthosts, rand_one_dut_hostname, tbinfo):
+    """
+    Setup/teardown
+    Args:
+        duthost: DUT.
+        golden_config_exists_on_dut: Check if golden config exists on DUT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    if duthost.is_multi_asic:
+        pytest.skip("Skip test on multi-asic platforms as it is designed for single asic.")
+
+    topo_type = tbinfo["topo"]["type"]
+    if topo_type in ["m0", "mx"]:
```

Comment:
Hi @yaqiangz , please check if the constraint is needed here.

Reply:
Yes, it is needed. Because I found you invoke config_reload with safe_reload=True, it will invoke pfcwd check depends on this whether it's enabled https://github.com/wen587/sonic-mgmt/blob/bd504802a3c76033d34b4087c7f0e2e84c92cf53/tests/common/config_reload.py#L198